File: JohanPersonnal/PHP/ServerCode/json_helper/SetupGame.php
<?php
/**
 * Created by PhpStorm.
 * User: johan
 * Date: 05.11.2017
 * Time: 19:05
 */

class SetupGame
{

    var $leaderID;
    var $race1;
    var $race2;
    var $gameID;
    var $mapID;
    var $opponentID;
    var $gamemode;


    /**
     * Set the setup
     * @param $race1
     * @param $race2
     * @param $mapID
     * @param $leaderID
     * @param $opponentID
     * @param $gameID
     * @param $gamemode
     */
    function set_setup($race1, $race2, $mapID, $leaderID, $opponentID, $gameID, $gamemode){


        $this->leaderID = $leaderID;
        $this->race1 = $race1;
        $this->gamemode = $gamemode;
        $this->gameID= $gameID;
        $this->opponentID = $opponentID;
        $this->race2 = $race2;
        $this->mapID = $mapID;

    }

    public function set_fromlobby($lobby)
    {
        $this->set_setup(-1,-1,-1,$lobby->leaderID,$lobby->opponentID,$lobby->gameID,$lobby->gamemode);
    }
}

File: Game/app/src/main/java/edu/iastate/cs/proj_309_vc_b_4/game/User/SetupGame.java
package edu.iastate.cs.proj_309_vc_b_4.game.User;

import edu.iastate.cs.proj_309_vc_b_4.game.Activities.playmenus.GameMode;

/**
 * a more precise lobby.
 * With extra information.
 * Created by johan on 05.11.2017.
 */

public class SetupGame extends Lobby {
    private final int race1;
    private final int race2;
    private final int mapID;

    /**
     * Return the setup of the game.
     * @param leaderID
     * @param race1 race of user 1  (leader)
     * @param race2 race of user 2
     * @param gameID the id of the game
     * @param mapID map ID.
     * @param opponentID the id of the opponent ( as the current user knows his own id)
     * @param gameMode the game mode selected
     */
    public SetupGame(int leaderID, int race1, int race2, int gameID, int mapID, int opponentID, GameMode gameMode){
            super(leaderID, "", gameMode, gameID, opponentID, "");
            this.race1 = race1;
            this.race2 = race2;
            this.mapID = mapID;
    }

    public int getMapID() {
        return mapID;
    }
}
